File: src/data_structure/linked_list/insert_at_position.cpp
/*
  Insert Node at a given position in a linked list 
  The linked list will not be empty and position will always be valid
  First element in the linked list is at position 0
  Node is defined as 
  struct Node
  {
     int data;
     struct Node *next;
  }
*/
Node* InsertNth(Node *head, int x, int position)
{
    int i;
    Node *prev, *cur;
    Node *n = (struct Node *)malloc(sizeof(struct Node));
    n->data = x;
    if (position == 0)
    {
        n->next = head;
        return n;
    }
    if (head == NULL)
    {
        n->next = NULL;
        return n;
    }
    cur = head;
    i=1;
    do
    {
        prev = cur;
        cur = cur->next;
    } while(cur != NULL && i++ < position);
    prev->next = n;
    n->next = cur;
    return head;
}

File: src/data_structure/linked_list/insert_at_position.java
/*
  Insert Node at the end of a linked list 
  head pointer input could be NULL as well for empty list
  Node is defined as 
  class Node {
     int data;
     Node next;
  }
*/
    // This is a "method-only" submission. 
    // You only need to complete this method. 

Node InsertNth(Node head, int data, int position) {
    Node n = new Node();
    n.data = data;
    if (head == null) {
        n.next = null;
        return n;
    }
    if (position == 0) {
        n.next = head;
        return n;
    }
    Node prev = head;
    Node cur = head;
    int i = 1;
    do {
        prev = cur;
        cur = cur.next;
    } while(cur != null && i++ < position);
    prev.next = n;
    n.next = cur;
    return head;
}

